nefertari_es/documents.py

```diff
@@ -16,27 +16,85 @@
     drop_reserved_params,
     split_strip,
     )
-from .meta import RegisteredDocMeta
-from .fields import ReferenceField
+from .meta import BackrefGeneratingDocMeta
+from .fields import ReferenceField, IdField
 
 
-@add_metaclass(RegisteredDocMeta)
+@add_metaclass(BackrefGeneratingDocMeta)
 class BaseDocument(DocType):
 
+    _public_fields = None
+    _auth_fields = None
+    _hidden_fields = None
     _nested_relationships = ()
 
-    # XXX implement auth and public fields?
-    _auth_fields = None
-    _public_fields = None
+    def __init__(self, *args, **kwargs):
+        super(BaseDocument, self).__init__(*args, **kwargs)
+        self._sync_id_field()
+
+    def _sync_id_field(self):
+        """ Copy meta["_id"] to IdField. """
+        if self.pk_field_type() is IdField:
+            pk_field = self.pk_field()
+            if not getattr(self, pk_field, None) and self._id:
+                setattr(self, pk_field, str(self._id))
+
+    def __getattr__(self, name):
+        if name == '_id' and 'id' not in self.meta:
+            return
```

Comment:
I mean that if you have a function whose return value is going to be used you should `return None` rather than just `return` to make it clear that you aren't just exiting the function but are returning a `None` value to the caller. This is a question of style.